esrally/utils/io.py

```diff
@@ -274,24 +285,54 @@ def decompress(zip_name, target_directory):
     this function.
     """
     path_without_extension, extension = splitext(zip_name)
-    filename = basename(path_without_extension)
     if extension == ".zip":
         _do_decompress(target_directory, zipfile.ZipFile(zip_name))
     elif extension == ".bz2":
-        _do_decompress_manually(target_directory, filename, bz2.open(zip_name))
+        decompressor_args = ["pbzip2", "-d", "-k", "-m10000", "-c"]
+        decompressor_lib = bz2.open
+        _do_decompress_manually(target_directory, zip_name, decompressor_args, decompressor_lib)
     elif extension == ".gz":
-        _do_decompress_manually(target_directory, filename, gzip.open(zip_name))
+        decompressor_args = ["pigz", "-d", "-k", "-c"]
+        decompressor_lib = gzip.open
+        _do_decompress_manually(target_directory, zip_name, decompressor_args, decompressor_lib)
     elif extension in [".tar", ".tar.gz", ".tgz", ".tar.bz2"]:
         _do_decompress(target_directory, tarfile.open(zip_name))
     else:
         raise RuntimeError("Unsupported file extension [%s]. Cannot decompress [%s]" % (extension, zip_name))
 
 
-def _do_decompress_manually(target_directory, filename, compressed_file):
+def _do_decompress_manually(target_directory, filename, decompressor_args, decompressor_lib):
+    decompressor_bin = decompressor_args[0]
+    base_path_without_extension = basename(splitext(filename)[0])
+
+    if is_executable(decompressor_bin):
+        if _do_decompress_manually_external(target_directory, filename, base_path_without_extension, decompressor_args):
+            return
+    else:
+        console.warn(f"{decompressor_bin} not found in PATH. Using standard library, decompression will take longer.")
+
+    _do_decompress_manually_with_lib(target_directory, filename, decompressor_lib(filename))
+
+
+def _do_decompress_manually_external(target_directory, filename, base_path_without_extension, decompressor_args):
+    with open(os.path.join(target_directory, base_path_without_extension), "wb") as new_file:
+        try:
+            subprocess.run(decompressor_args + [filename], stdout=new_file, stderr=subprocess.PIPE, check=True)
+        except subprocess.CalledProcessError as err:
+            logger = logging.getLogger(__name__)
+            console.warn(f"Failed to decompress [{filename}] with [{err.cmd}]. Error [{err.stderr}]. Falling back to standard library.",
+                         logger=logger.warning)
```

Comment:
This produces:

```
[ERROR] Cannot race. Error in track preparator ('function' object has no attribute 'warning')
```

you just need to provide the `logger` instead of `logger.warning`.

Alternatively, I also wonder whether we should either fail hard here because something is clearly wrong with the `pbzip` binary or instead just write it to the logs and continue with the stdlib implementation. Here is the output from one of my tests (after I changed `logger=logger.warning` to `logger=logger`):

```
[INFO] Decompressing track data from [/Users/daniel/.rally/benchmarks/data/http_logs/documents-191998-1k.json.bz2] to [/Users/daniel/.rally/benchmarks/data/http_logs/documents-191998-1k.json] ... [WARNING] Failed to decompress [/Users/daniel/.rally/benchmarks/data/http_logs/documents-191998-1k.json.bz2] with [['pbzip2', '/Users/daniel/.rally/benchmarks/data/http_logs/documents-191998-1k.json.bz2']]. Error [b'pbzip2: *ERROR: Input file [/Users/daniel/.rally/benchmarks/data/http_logs/documents-191998-1k.json.bz2] already has a .bz2 extension!  Skipping...\n-------------------------------------------\n']. Falling back to standard library.
```

Reply:
Switched to logging only in https://github.com/elastic/rally/pull/947/commits/476e3fd9051fa3db43cf6e85ec9a29b0f704f3de